x-pack/plugins/ml/server/lib/check_privileges/check_privileges.ts

```diff
@@ -37,9 +35,9 @@ export function privilegesProvider(
     const privileges = getDefaultPrivileges();
 
     const upgradeInProgress = await isUpgradeInProgress();
-    const securityDisabled = licenseCheckResult.isSecurityDisabled;
-    const license = checkLicense(licenseCheckResult);
-    const isPlatinumOrTrialLicense = license.licenseType === LICENSE_TYPE.FULL;
+    const isSecurityEnabled = mlLicense.isSecurityEnabled();
+
+    const isPlatinumOrTrialLicense = mlLicense.isFullLicense();
```

Comment:
Should this be renamed to `isFullLicense` as isn't this check for platinum, trial, or enterprise?

Reply:
unfortunately `isPlatinumOrTrialLicense` is part of our `ml_capabilities` endpoint response which is used by other plugins.
If we want to change this i'd suggest doing it in a followup PR which also updates the code in the plugins which consume it.
we have the plan to remove `ml_capabilities` in favour of kibana's built in capabilities information, so maybe it's worth waiting until then.

Reply:
Agree that it makes sense to leave this as `isPlatinumOrTrialLicense` and wait until we look at removing `ml_capabilities` in favour of Kibana's built-in capabilities.